Common/Product/TestAdapter/VsProjectExtensions.cs

```diff
@@ -87,6 +94,21 @@ public static bool IsTestProject(this IVsProject project, Guid projectGuid)
             return (projectTypeGuids.IndexOf(projectGuid.ToString(), StringComparison.OrdinalIgnoreCase) >= 0);
         }
 
+        public static bool IsTestAdapaterEnabled()
```

Comment:
I just made some changes in my VS PR to cache this value, since "IsTestProject" gets called many times during any single TestAdapter operation.  We don't really need to hit the registry every time.   You don't need to make a similar change in NTVS, up to you.